src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Col, Button, Alert } from 'patternfly-react';
+
+import { FormFactory } from '../Form/FormFactory';
+import { createBaremetalHost } from '../../k8s/request';
+import { TinyInlineLoading } from '../Loading/Loading';
+
+const formFields = {
+  name: {
+    id: 'name',
+    title: 'Name',
+    required: false,
+  },
+  controller: {
+    id: 'controller',
+    title: 'Management Controller Address',
+    required: true,
+  },
+  username: {
+    id: 'username',
+    title: 'Username',
+    required: true,
+  },
+  password: {
+    id: 'password',
+    title: 'Password',
+    type: 'password',
+    required: true,
+  },
+};
+
+const formText = `Specify the host's name, management controller IP address,
```

Comment:
nit: please create `strings.js` file for that and use `UPPER_CASE` constant convention.

Reply:
Done